### ISSUE TITLE
Support POST requests within online routing

### ISSUE THREAD

**Issue by joker234** [osmandapp/OsmAnd#12995]
# 🚀 feature request

### Description

tl;dr: are you willing to merge a PR supporting POST requests in the online routing integration.

When doing #12573 (thx @vshcherb for reviewing and merging that so quickly :bow:), I found out that the OsmAnd online routing integration only supports GET requests. Since openrouteservice only has a very limited featureset through GET and supports a lot of features only through POST, we would be interested if the OsmAnd team would be willing to integrate POST request and especially **merge a Pull Request integrating POST into this part**, in general.

### Describe the solution you'd like

A brief discussion if it's worth it to put some effort in a PR and how the chances are of one being merged (if it complies with the standards you have, for sure).

### Describe alternatives you've considered

The only alternative I see is to stick with the half-functioning openrouteservice integration there is now in OsmAnd. One example which wouldn't be working with GET requests are intermediate stops.

**Comment by joker234** [comment]
>  Another option would be to default to POST or GET request in the specific Engine and not have the setting in the config (easiest :).

I would vote for that. Imho it's the responsibility of the Type provider/developer (Graphhopper, OSRM, openrouteservice) to implement that and choose if they want to use POST or GET. I would only allow the user to pick an option if there are differences in the supported featureset and one is not a sub-featureset of another (as GET often supports a subset of POST within APIs). I wouldn't say the pick for the user needs to be implemented at the moment.

**Comment by vshcherb** [comment]
@TheGreatRefrigerator I think if you have PR prepared, it would be quite easy to accept it. This is rather technical setting which could be hidden unless Online Routing is activated

**Comment by TheGreatRefrigerator** [comment]
@vshcherb sorry for the late response and thank you for the feedback. Actually someone else will do the PR. Currently planned for between the years. We can discuss further details directly in the PR then i guess :)
